core/frontend/src/main/scala/io/udash/Application.scala

```diff
@@ -39,9 +43,20 @@ class Application[HierarchyRoot <: GState[HierarchyRoot] : ClassTag : ImmutableV
       .recover { case ex: Throwable => handleRoutingFailure(ex) }
   }
 
-  def handleRoutingFailure(ex: Throwable): Unit = {
+  def registerRoutingFailureListener(listener: RoutingFailureListener): Registration = {
```

Comment:
Document and test this public method. The users need to know the ordering and execution guarantees. What happens to other listeners if one of them throws an exception?